python/lsst/sims/maf/metricBundles/metricBundle.py

```diff
@@ -119,9 +119,16 @@ def _setupMetricValues(self):
         # Can't store healpix slicer mask values in an int array.
         if dtype == 'int':
             dtype = 'float'
-        self.metricValues = ma.MaskedArray(data = np.empty(len(self.slicer), dtype),
-                                            mask = np.zeros(len(self.slicer), 'bool'),
-                                            fill_value= self.slicer.badval)
+        # Check if we have a 2D slicer
+        if 'bins' in self.slicer.slicePoints.keys():
```

Comment:
It may be beyond the scope of htis ticket, but would it be worthwhile to have a look at what I ended up doing in the moving object metricBundle / moving object slicer? 
Basically, I let the slicer have a shape .. the iteration over this shape is a little clunky at present but maybe if all slicers had shapes (i.e. more than one possible dimension) we could make this a little smoother?